packages/fx-core/src/component/generator/error.ts

```diff
@@ -25,6 +25,30 @@ export class UnzipError extends BaseComponentInnerError {
   }
 }
 
+export class DownloadSampleNetworkError extends BaseComponentInnerError {
+  constructor(url: string) {
+    super(
+      errorSource,
+      "SystemError",
```

Comment:
I think the download error caused by network issue and api limit is "User error" because they can be recovered after user retry it.